routes/api/v1/signup.js

```diff
@@ -0,0 +1,30 @@
+var express = require('express');
+var router = express.Router();
+
+router.post('/', (req, res, next) => {
+  var db = req.app.get('db')
+  var user = req.body
+
+  if(user.username && user.password) {
+    db('users').where({username: user.username})
+    .then( (existingUser) => {
+      if(existingUser[0]) {
+        return res.json({errorMessage: 'Username already exists'})
+      } else {
+        //hash password with bcrypt before saving
+        db('users').insert(user)
+        .then( () => {
+          //generate JWT
+          //return JWT
+          return res.json()
+        });
+      }
+    });
+    }else{
+      var err = new Error('Not found');
+      err.status = 400;
+      next(err);
+    }
```

Comment:
Yes, Rails is definitely great in that it makes a lot of decisions for you so you don't have to; however, that doesn't really help you when you need to do something outside of Rails. It's not really the fault of Rails, it's their job to do that, so as a developer, you simply work through these things as they come up. We've all been there.

Happy holidays to you as well.